src/main/resources/assets/js/components/AppComponent.jsx

```diff
@@ -3,8 +3,15 @@
 var React = require("react/addons");
 
 var AppComponent = React.createClass({
+  propTypes: {
+    router: React.PropTypes.object.isRequired
+  },
+
   onClick: function () {
-    this.props.onClick(this.props.model);
+    this.props.router.navigate(
+      "#apps" + this.props.model.get("id"),
```

Comment:
This is dependent on using hashes, it would be nice if we left that up to the router to handle whether it is fragments or push state.